BenchmarkTests/Benchmarks/Sources/Benchmarks.swift

```diff
@@ -0,0 +1,133 @@
+                applicationVersion: String,
+                sdkVersion: String,
+                deviceModel: String,
+                osName: String,
+                osVersion: String,
+                run: String,
+                scenario: String,
+                branch: String
+            ) {
+                self.applicationIdentifier = applicationIdentifier
+                self.applicationName = applicationName
+                self.applicationVersion = applicationVersion
+                self.sdkVersion = sdkVersion
+                self.deviceModel = deviceModel
+                self.osName = osName
+                self.osVersion = osVersion
+                self.run = run
+                self.scenario = scenario
+                self.branch = branch
+            }
+        }
+
+        var clientToken: String
+        var apiKey: String
+        var context: Context
+
+        public init(
+            clientToken: String,
+            apiKey: String,
+            context: Context
+        ) {
+            self.clientToken = clientToken
+            self.apiKey = apiKey
+            self.context = context
+        }
+    }
+    
+    /// Configure OpenTelemetry metrics meter and start measuring Memory.
+    ///
+    /// - Parameter configuration: The Benchmark configuration.
+    public static func metrics(with configuration: Configuration) {
+        let loggerProvider = LoggerProviderBuilder()
+            .build()
+
+        let metricExporter = MetricExporter(
+            configuration: MetricExporter.Configuration(
+                apiKey: configuration.apiKey,
+                version: instrumentationVersion
+            )
+        )
+
+        let meterProvider = MeterProviderBuilder()
+            .with(pushInterval: 10)
+            .with(processor: MetricProcessorSdk())
+            .with(exporter: metricExporter)
+            .with(resource: Resource())
+            .build()
+
+        let logger = loggerProvider
+            .loggerBuilder(instrumentationScopeName: instrumentationName)
+            .build()
+
+        let meter = meterProvider.get(
+            instrumentationName: instrumentationName,
+            instrumentationVersion: instrumentationVersion
+        )
+
+        let labels = [
+            "device_model": configuration.context.deviceModel,
+            "os": configuration.context.osName,
+            "os_version": configuration.context.osVersion,
+            "run": configuration.context.run,
+            "scenario": configuration.context.scenario,
+            "application_id": configuration.context.applicationIdentifier,
+            "sdk_version": configuration.context.sdkVersion,
+            "branch": configuration.context.branch,
+        ]
+
+        _ = meter.createDoubleObservableGauge(name: "ios.benchmark.memory") { metric in
+            do {
+                let mem = try Memory.footprint()
+                metric.observe(value: mem, labels: labels)
```

Comment:
question/ How does it work? We create Gauge metric ✅ but when / how frequently will this closure be called?

Reply:
It is based on [Asynchronous Gauge](https://opentelemetry.io/docs/specs/otel/metrics/api/#asynchronous-gauge) from otel specs where _Callback functions will be called only when the Meter is being observed._
The meter is observed when there is a push, [in our case it will be every 10s](https://github.com/DataDog/dd-sdk-ios/pull/1993/files#diff-b5ae3e89d1c4c79036d3ea45588a247fff697d5ccf90f172c0be2856a4542e79R92)

Reply:
Got it, thanks!